classes/query/QueryDAO.inc.php

```diff
@@ -0,0 +1,314 @@
+		$query->setSequence($row['seq']);
+
+		HookRegistry::call('QueryDAO::_fromRow', array(&$query, &$row));
+		return $query;
+	}
+
+	/**
+	 * Get a new data object
+	 * @return DataObject
+	 */
+	function newDataObject() {
+		return new Query();
+	}
+
+	/**
+	 * Insert a new Query.
+	 * @param $query Query
+	 */
+	function insertObject($query) {
+		$this->update(
+			'INSERT INTO queries (assoc_type, assoc_id, stage_id, closed, seq)
+			VALUES (?, ?, ?, ?, ?)',
+			array(
+				(int) $query->getAssocType(),
+				(int) $query->getAssocId(),
+				(int) $query->getStageId(),
+				(int) $query->getIsClosed(),
+				(float) $query->getSequence(),
+			)
+		);
+		$query->setId($this->getInsertId());
+		return $query->getId();
+	}
+
+	/**
+	 * Adds a participant to a query.
+	 * @param $queryId int Query ID
+	 * @param $userId int User ID
+	 */
+	function insertParticipant($queryId, $userId) {
+		$this->update(
+			'INSERT INTO query_participants
+			(query_id, user_id)
+			VALUES
+			(?, ?)',
+			array(
+				(int) $queryId,
+				(int) $userId,
+			)
+		);
+	}
+
+	/**
+	 * Removes a participant from a query.
+	 * @param $queryId int Query ID
+	 * @param $userId int User ID
+	 */
+	function removeParticipant($queryId, $userId) {
+		$this->update(
+			'DELETE FROM query_participants WHERE query_id = ? AND user_id = ?',
+			array((int) $queryId, (int) $userId)
+		);
+	}
+
+	/**
+	 * Removes all participants from a query.
+	 * @param $queryId int Query ID
+	 */
+	function removeAllParticipants($queryId) {
+		$this->update(
+			'DELETE FROM query_participants WHERE query_id = ?',
+			(int) $queryId
+		);
+	}
+
+	/**
+	 * Retrieve all participant user IDs for a query.
+	 * @param $queryId int Query ID
+	 * @param $userId int User ID to restrict results to
+	 * @return array
+	 */
+	function getParticipantIds($queryId, $userId = null) {
+		$params = array((int) $queryId);
+		if ($userId) $params[] = (int) $userId;
+		$result = $this->retrieve(
+			'SELECT	user_id
+			FROM	query_participants
+			WHERE	query_id = ?' .
+			($userId?' AND user_id = ?':''),
+			$params
+		);
+		$userIds = array();
+		while (!$result->EOF) {
+			$row = $result->getRowAssoc(false);
+			$userIds[] = $row['user_id'];
+			$result->MoveNext();
+		}
+		$result->Close();
+		return $userIds;
+	}
+
+	/**
+	 * Update an existing Query.
+	 * @param $query Query
+	 */
+	function updateObject($query) {
+		$this->update(
+			'UPDATE	queries
+			SET	assoc_type = ?,
+				assoc_id = ?,
+				stage_id = ?,
+				closed = ?,
+				seq = ?
+			WHERE	query_id = ?',
+			array(
+				(int) $query->getAssocType(),
+				(int) $query->getAssocId(),
+				(int) $query->getStageId(),
+				(int) $query->getIsClosed(),
+				(float) $query->getSequence(),
+				(int) $query->getId()
+			)
+		);
+	}
+
+	/**
+	 * Delete a submission file query.
+	 * @param $query Query
+	 */
+	function deleteObject($query) {
+		$this->deleteById($query->getId());
+	}
+
+	/**
+	 * Delete a submission file query by ID.
```

Comment:
same from above, I think we can avoid specifying which object the query is assigned to here...